### DISCUSSION TITLE
godog fails in gitlab pipeline godog@v0.12.5/internal/builder/builder.go:243:15: undefined: os.ReadFile

### DISCUSSION THREAD

**Discussion by ScottiePower** [cucumber/godog#508] [General]
Hello,
I am trying to add godog tests to my gitlab pipeline but consistently get an error in the builder.go with os.ReadFile.
Below is an extract for the gitlab runner. I can not find any other reference to this issue. 
Can any one please help.

$ go version
go version go1.15.15 linux/amd64
$ GO111MODULE=on go get github.com/cucumber/godog/cmd/godog@v0.12.0
go: downloading github.com/cucumber/godog v0.12.0
go: found github.com/cucumber/godog/cmd/godog in github.com/cucumber/godog v0.12.0
go: downloading github.com/spf13/cobra v1.1.1
go: downloading github.com/cucumber/gherkin-go/v19 v19.0.3
go: downloading github.com/cucumber/messages-go/v16 v16.0.1
go: downloading github.com/hashicorp/go-memdb v1.3.0
go: downloading github.com/gofrs/uuid v4.0.0+incompatible
go: downloading github.com/hashicorp/go-immutable-radix v1.3.0
go: downloading github.com/inconshreveable/mousetrap v1.0.0
go: downloading github.com/hashicorp/golang-lru v0.5.4
$ godog version
Godog version is: v0.12.0
$ pwd
/builds/****/******
$ cd *****
$ go test -v
go: downloading github.com/cucumber/godog v0.12.5
go: downloading github.com/hashicorp/go-immutable-radix v1.3.1
# github.com/cucumber/godog/internal/builder
/root/go/pkg/mod/github.com/cucumber/godog@v0.12.5/internal/builder/builder.go:243:15: undefined: os.ReadFile

**Comment by vearutop** [comment]
Hi, `go1.15` is too old to be supported, please use at least `go1.16`.

**Reply by ScottiePower** [reply]
Thank you, will try that.

**Reply by Anatol-Misnik** [reply]
> Hi, `go1.15` is too old to be supported, please use at least `go1.16`.

Hi. What about go1.18.1 - is not too recent version?)

**Reply by vearutop** [reply]
go1.18+ is ok in general, though it has an issue with step function names in formatter.
https://github.com/cucumber/godog/pull/466#issuecomment-1071554522
